Fix NPEs cause by null times

alexabird opened the pull request.

alexabird committed:
Fix NPE thrown when start and end times are null

dcruver commented on src/main/java/org/codice/itest/TestExecutorTask.java:
```diff
@@ -59,19 +59,15 @@ public void run() {
             this.notify(testResultFactory.pass(testName, beforeTest, afterTest));
         } catch (AssertionError e) {
-            if(beforeTest!=null && afterTest==null)
-                afterTest = Instant.now();
```
:pencil: We could declare this in `notify()` and drop the conditional.

dcruver commented on src/main/java/org/codice/itest/TestExecutorTask.java:
```diff
@@ -59,19 +59,15 @@ public void run() {
             this.notify(testResultFactory.fail(testName, e.getMessage(), beforeTest, afterTest));
         } catch (Throwable t) {
-            if(beforeTest!=null && afterTest==null)
-                afterTest = Instant.now();
```
:memo:  We could declare this in `notify()` and drop the conditional.

dcruver commented on src/main/java/org/codice/itest/TestExecutorTask.java:
```diff
@@ -59,19 +59,15 @@ public void run() {
                 diagnosticTest.cleanup();
             } catch (Exception e) {
-                if(beforeTest!=null && afterTest==null)
-                    afterTest = Instant.now();
```
:memo:  We could declare this in `notify()` and drop the conditional.

dcruver reviewed (commented):
:exclamation: Even with this change in place, if `setup()` throws an exception, we're going to get an NPE. I'd suggest we switch lines 55 and 56 and move all occurrences of `afterTest = Instant.now()` into `notify()`. Then you can remove the conditionals as well.

alexabird commented:
@dcruver I made some changes that make this cleaner and I think address your core concerns. 
* Called `Instant.now()` for the endTime paramter in the `TestResult` initializers, instead of moving `afterTest = Instant.now()` to the `notify()` method (would require more changes than just that line move since the `TestResult` initializers need the `endTime`)
* Moved `setup()` within the timed part of the test, and also moved `cleanup()` into timed portion.
